src/utility/svd.jl

```diff
@@ -324,6 +324,12 @@ function ChainRulesCore.rrule(
 ) where {F<:Union{IterSVD,FixedSVD},R<:Union{GMRES,BiCGStab,Arnoldi},B}
     U, S, V, info = PEPSKit.tsvd(f, alg; trunc, p)
 
+    # update rrule_alg tolerance to be compatible with smallest singular value
+    rrule_alg = alg.rrule_alg
+    smallest_sval = minimum(minimum(abs.(diag(b))) for (_, b) in blocks(S))
```

Comment:
```suggestion
    smallest_sval = minimum(((_, b),) -> minimum(abs ∘ diag, b), blocks(S)) 
```
Is the `abs` really necessary though? Given that `S` comes from an SVD, that should already be real?